nixos/modules/programs/steam.nix

```diff
@@ -44,7 +44,14 @@ in {
 
     hardware.steam-hardware.enable = true;
 
-    environment.systemPackages = [ steam steam.run ];
+    environment.systemPackages = [
```

Comment:
Why not just inline this overriding into the derivation for the FHSEnv?

Reply:
That's because (at least, i think) the `config` attributes (like `config.hardware.nvidia.package`) can't be referred to from within packages

Reply:
Yeah, you can't do that, but if the ICD files are similar/identical for all Nvidia driver packages, then you could use a generic one in the FHS.

Reply:
I don't think the ICD files are similar enough to do that; my reasoning is a bit anecdotal as im not entirely sure of how this works, but using a different version of the nvidia driver (nvidiaBeta or nvidiaVulkanBeta) caused proton/dxvk to throw errors and not work for me.

Reply:
I just checked, and yeah, they look similar but they point to different store paths, so it wouldn't work.

Reply:
What happens if I'm not using Nvidia on my system? Will this then pull in Nvidia drivers even though I don't use them?

Reply:
im actually not sure how the fhsenv gets the ICD files to use. If i knew that, i think i could come up with a better/cleaner solution.
as for pulling in nvidia, i think it will... i'm unsure how else to handle it other than a manual switch to toggle it on, but that then adds another toggle on top of enabling the steam module for nvidia users, or a switch to toggle it off for non-nvidia users.

Reply:
From what I can tell, the `icd` files you have are driver-specific, and are currently in `/run/opengl-driver/share/...`.
I took a look at this, and in the FHS that is made, we still have access to this folder, so you might be able to just add all the `icd` files in `/run/opengl-driver/share/vulkan/icd.d/` to `VK_ICD_FILENAMES` for Steam. I am also not sure where the current `icd` files in `/usr/` come from.

In essence, taking the existing `VK_ICD_FILENAMES`, and appending to it itself, but with `/usr` replaced with `/run/opengl-driver`, might make it all work magically.

Reply:
most of the icd files come from the `mesa.drivers` package except for the nvidia files which i have no clue where they're _supposed_ to come from.
Getting the icd file from `/run/opengl-driver/` also isnt good enough because it does not include the 32 bit driver which is necessary for 32 bit games.